### PR TITLE
Create recurrence general configuration

### PR DESCRIPTION
**Pull request by GabrielDeveloper**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Issues         | https://github.com/mundipagg/plug-team/issues/203
| What?         | Create configuration of Recurrence and improve UX
| Why?          | To save the generals configurations of recurrence feature
| How?          | We create a new tab on settings to configure the recurrence and improve UX on templates screen

captura de tela de 2018-06-26 10-19-51

captura de tela de 2018-06-26 10-21-16

captura de tela de 2018-06-26 10-22-35

### CODE REVIEW

File: admin/view/script/mundipagg.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
+function removeRow(e) {
```
Comment (ianlessa) [inline admin/view/script/mundipagg.js]: Colocar um nome que descreva a funcionalidade deste arquivo.

File: system/library/mundipagg/src/Helper/AdminMenu.php [hunk 2]
```diff
@@ -31,6 +46,28 @@ public function getMenu()
...
+        if ($this->recurrenceSettings->isSingleRecurrenceEnable()) {
+            $result['children'][] = $this->getMenuChildren('Single');
+        }
+
+        if ($this->recurrenceSettings->isSubscriptionByPlanEnable()) {
+            $result['children'][] = $this->getMenuChildren('Plans');
+        }
+
+        return $result;
+
```
Comment (michelpl) [inline system/library/mundipagg/src/Helper/AdminMenu.php]: Linha extra

File: system/library/mundipagg/src/Controller/Recurrence/Templates.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        $this->data['panelPlanFrequency'] = 'extension/payment/mundipagg/recurrence/templates/panelPlanFrequency.twig';
+
+        $this->data['formSingle'] = 'extension/payment/mundipagg/recurrence/templates/form_single.twig';
+        $this->data['panelSingleFrequency'] = 'extension/payment/mundipagg/recurrence/templates/panelSingleFrequency.twig';
+
+        $path = 'extension/payment/mundipagg/';
+        $this->data['formBase'] = $path . 'recurrence/templates/form_base.twig';
+        $this->render('templates/create');
+    }
+
```
Comment (michelpl) [inline system/library/mundipagg/src/Controller/Recurrence/Templates.php]: linha extra

File: system/library/mundipagg/src/Controller/Recurrence/Recurrence.php [hunk 4]
```diff
@@ -43,4 +47,11 @@ public function render($path)
             )
         );
     }
+
+    protected function chargeRecurrenceSettings()
+    {
+        $this->recurrenceSettings = new RecurrenceSettings($this->openCart);
+        $this->data['recurrenceSettings'] = $this->recurrenceSettings->getAllSettings();
+    }
+
```
Comment (michelpl) [inline system/library/mundipagg/src/Controller/Recurrence/Recurrence.php]: Linha extra

File: system/library/mundipagg/src/Controller/Events.php [hunk 5]
```diff
@@ -111,30 +111,76 @@ public function columnLeftEntry($data)
...
+       $data['tab_design'] = $data['tab_design'] . $productFormTemplate;
+       $data['footer'] = $data['footer'] . $productFormTabContentTemplate;
+
+       foreach ($data as $key => $value) {
+           $this->template->set($key, $value);
+       }
+
+       return $this->template;
+    }
+
```
Comment (michelpl) [inline system/library/mundipagg/src/Controller/Events.php]: Linha extra

File: system/library/mundipagg/src/Controller/Events.php [hunk 6]
```diff
@@ -111,30 +111,76 @@ public function columnLeftEntry($data)
...
+       }
+
+       return $this->template;
     }
-}
\ No newline at end of file
+
+    public function handleRecurrenceSingleTab($data)
+    {
+
```
Comment (michelpl) [inline system/library/mundipagg/src/Controller/Events.php]: Linha extra

File: admin/view/template/extension/payment/mundipagg/recurrence/templates/panelSingleFrequency.twig [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+            </div><!-- /input-group -->
+        </div>
+        <div class="col-sm-3">
+            <button type="submit" form="form-product" title="" class="btn btn-primary btn-add-frequency" data-original-title="Save"><i class="fa fa-plus-circle"></i> </button>
+        </div>
+    </div>
+</div>
+
+<div class="panels">
+
```
Comment (michelpl) [inline admin/view/template/extension/payment/mundipagg/recurrence/templates/panelSingleFrequency.twig]: Linha extra

**Pull request merged**